content/src/ports/activeEntities.ts

```diff
@@ -0,0 +1,169 @@
+import { Deployment, Entity, EntityId, PartialDeploymentHistory, Pointer } from 'dcl-catalyst-commons'
+import LRU from 'lru-cache'
+import { EnvironmentConfig } from '../Environment'
+import { getDeployments } from '../service/deployments/deployments'
+import { AppComponents } from '../types'
+
+export type ActiveEntities = {
+  withPointers(...pointers: Pointer[]): Promise<Entity[]>
+  withIds(...entityIds: EntityId[]): Promise<Entity[]>
+  inactivate(...pointers: Pointer[]): void
+  activate(pointer: Pointer, entity: Entity): void
+}
+
+const mapDeploymentsToEntities = (history: PartialDeploymentHistory<Deployment>): Entity[] => {
+  return history.deployments.map(
+    ({ entityVersion, entityId, entityType, pointers, entityTimestamp, content, metadata }) => ({
+      version: entityVersion,
+      id: entityId,
+      type: entityType,
+      pointers,
+      timestamp: entityTimestamp,
+      content: content?.map(({ key, hash }) => ({ file: key, hash })),
+      metadata
+    })
+  )
+}
+
+export const createActiveEntitiesComponent = (
+  components: Pick<AppComponents, 'database' | 'env' | 'logs' | 'metrics'>
+): ActiveEntities => {
+  const logger = components.logs.getLogger('ActiveEntities')
+  const cache = new LRU<EntityId, Entity>({
+    max: components.env.getConfig(EnvironmentConfig.ENTITIES_CACHE_SIZE)
+  })
+  const entityIdByPointers = new Map<Pointer, EntityId>()
+
+  const reportCacheAccess = (entity: Entity, result: 'hit' | 'miss') => {
+    components.metrics.increment('dcl_entities_cache_accesses_total', {
+      entity_type: entity.type,
+      result: result
+    })
+  }
+
+  const cacheEntity = (entity: Entity): void => {
+    cache.set(entity.id, entity)
+  }
+
+  const invalidateEntity = (entityId: EntityId): void => {
+    const entity = cache.get(entityId)
+    if (entity) {
+      for (const pointer of entity.pointers) {
+        entityIdByPointers.delete(pointer)
+      }
+    }
+    cache.del(entityId)
+  }
+
+  const updateCacheAndReturnAsEntities = (deployments: PartialDeploymentHistory<Deployment>): Entity[] => {
```

Comment:
I love how function names are clear enough to not require any extra comment in the code